src/pages/ShoppingList.js

```diff
@@ -1,7 +1,41 @@
-import React from 'react';
+import React, { useState, useEffect } from 'react';
+import fb from '../lib/firebase';
 
-const ShoppingList = () => {
-	return <h1>Hello from the Shopping List</h1>;
+const ShoppingList = ({ token }) => {
+	const [shoppingListItems, setShoppingListItems] = useState([]);
+
+	//Thanks for sharing your code!!
+	useEffect(() => {
+		const db = fb.firestore();
+		const tokenRef = db.collection(token);
+		tokenRef
+			.orderBy('timeFrame', 'asc')
+			.get()
+			.then(querySnapshot => {
+				let fullCollection = [];
+				querySnapshot.forEach(doc => {
+					let documentData = doc.data();
+					let nameData = documentData.itemName;
+					nameData = nameData
+						.toLowerCase()
+						.replace(/[.,\/#!$%\^&\*;:{}=\-_`~()]/g, '')
+						.trim()
+						.replace(/\s{2,}/g, ' ');
+					fullCollection.push(nameData);
+				});
+				setShoppingListItems(fullCollection);
+			})
+			.catch(error => {
+				console.log('Error getting document:', error);
+			});
+	}, []);
+	return (
+		<ul>
+			{shoppingListItems.length > 0 && shoppingListItems.map(item => (
```

Comment:
It might be nice to add some sort of output for length of the shoppingListItems array is equal to 0, so the user knows when  the shopping list is empty versus the page not loading correctly, or still loading.